packages/react/src/animation/TextRevealAnimation/TextRevealAnimation.tsx

```diff
@@ -0,0 +1,72 @@
+import React, {useCallback, useEffect} from 'react'
+import clsx from 'clsx'
+import {useTextRevealAnimation} from './useTextRevealAnimation'
+
+import styles from './TextRevealAnimation.module.css'
+
+export function TextRevealAnimation({children, ...rest}) {
+  const [animationStarted, setAnimationStarted] = React.useState(false)
+  const [hasHeroWipeAnimated, setHasHeroWipeAnimated] = React.useState(false)
+  const {ref, lines} = useTextRevealAnimation(children)
+
+  const onLineAnimationEnd = useCallback(
+    (i: number) => {
+      if (lines && i === lines.length - 1) {
+        setHasHeroWipeAnimated(true)
+      }
+    },
+    [lines],
+  )
+
```

Comment:
I think this might be a case where `useCallback` is going to be less-performant than not using it. 

`onLineAnimationEnd` is only passed to a span, and so no other component renders are tied to the identity of this function, meaning there's no benefit in keeping the identity of the function the same across renders. All we're doing is adding the runtime overhead of `useCallback`, along with the memory footprint of memoizing the function. Both of those are admittedly tiny, but mentioning as all we're doing here is adding code that doesn't give any benefit.

I think I wrote this originally, so this is a comment to my past self more than anything 😅

Reply:
👍 yeah, fair points.. have removed the memoization. I think a similar issue is the recreation of the inline, anonymous function on each render, which would have been the one we wanted to memoize. Arguably negligible impact though, so will leave as-is.

Reply:
Great stuff, thanks 🙌

It's a bit counterintuitive, but the anonymous function will be recreated on every render regardless of whether we use `useCallback`.

Kent C Dodds' article breaks it down quite well ([link to relevant section](https://kentcdodds.com/blog/usememo-and-usecallback#why-is-usecallback-worse))